src/readers/tests/test.cpp

```diff
@@ -2,8 +2,8 @@
 
 #include <fstream>
 
-#include "../src/virtex_data_provider.h"
-#include "../src/virtex_reader.h"
+#include "../src/type1/type1_data_provider.h"
+#include "../src/type1/type1_eader.h"
 
 std::string sample =
```

Comment:
you call this file just `test`, but the new one is `type2_test`. Please change file names to meaningfull ones